Add content about LSP8 TokenId Format + Metadata differences (#766)
lukso-network/docs: docs/contracts/overview/DigitalAssets.md

</details>

### Setting metadata for one or multiple `tokenId`s

The function [`setDataBatchForTokenIds(bytes32[],bytes32[],bytes[])`](../../contracts/contracts/LSP8IdentifiableDigitalAsset/LSP8IdentifiableDigitalAsset.md#setdatabatchfortokenids) can be used to set multiple data key-value pairs at once for one or multiple tokenIds.

> This function is flexible enough to enable to:

Reviewer: ```suggestion
This function is flexible enough to set one or multiple [data key-value|(../../standards/lsp-background/erc725.md#erc725y-generic-data-keyvalue-store) pairs for:
```

IMO the bullet point list should only feature the differences to improve clarity. Also added a link for the data keys of ERC725Y.

> - **case 1:** either set multiple data key-value pairs for the same tokenId.
> - **case 2:** set data key-value pairs for different tokenIds.

Reviewer: ```suggestion
- a single `tokenId`
- different `tokenId` elements
```

- I think there is no need to keep the `either` as the bullet points do not finish the sentence.
- I would rather create separate menus / headings in order to keep the file page more organized.

Author: Done created different subheadings

> The parameters of `setDataBatchForTokenIds(bytes32[],bytes32[],bytes[])` will be as follow depending on the 2 cases:
> 
> **case 1: set 3 x different data key-value pairs for the same tokenId**
> 
> ```solidity

Reviewer: ```suggestion
To set [LSP4](../../standards/tokens/LSP4-Digital-Asset-Metadata.md) metadata, the parameters of the `setDataBatchForTokenIds(..)` function will vary depending on the usecase:

<Tabs>
  <TabItem value="single-token-id" label="Set metadata for single token ID">

:::info

The following code snippet will set three different key-value pairs for the same `tokenId`

:::

```solidity
```

As the page is pretty big, I would suggest menu bars. If you like the idea, don't forget to add the imports at the top of the file.

```
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
```

bytes32 constant _NFT_MARKET_PLACE_URLS__DATA_KEY = keccak256("NFTMarketplaceURLs");

bytes32 constant _TOKEN_ID_TO_SET = 0xcafecafecafecafecafecafecafecafecafecafecafecafecafecafecafecafe;

function setMultipleDataForSingleTokenId(
    ILSP8 lsp8Contract,
    bytes memory lsp4MetadataValue,
    bytes memory nftIconValue,
    bytes memory nftMarketPlaceURLsValue
) {
    bytes32[] memory tokenIdsToUpdate = new bytes32[](3);
    bytes32[] memory dataKeysToSet = new bytes32[](3);
    bytes[] memory dataValuesToSet = new bytes[](3);

    // we are setting 3 x data key-value pairs for the same tokenid
    tokenIdsToUpdate[0] = _TOKEN_ID_TO_SET;
    tokenIdsToUpdate[1] = _TOKEN_ID_TO_SET;
    tokenIdsToUpdate[2] = _TOKEN_ID_TO_SET;

    dataKeysToSet[0] = _LSP4_METADATA_KEY;
    dataKeysToSet[1] = _NFT_ICON_DATA_KEY;
    dataKeysToSet[2] = _NFT_MARKET_PLACE_URLS__DATA_KEY;

    dataValuesToSet[0] = lsp4MetadataValue;
    dataValuesToSet[1] = nftIconValue;
    dataValuesToSet[2] = nftMarketPlaceURLsValue;

    lsp8Contract.setDataBatchForTokenIds(
        tokenIdsToUpdate,
        dataKeysToSet,
        dataValuesToSet
    );
}
```

> 
> **case 2: set the LSP4 Metadata of 3 x different tokenIds**
> 
> ```solidity

Reviewer: ```suggestion

</TabItem>
<TabItem value="multiple-token-ids" label="Set metadata for multiple Token IDs">

:::info

The following code snippet will set three different key-value pairs for multiple `tokenId` elements.

:::

```solidity
```

bytes32 constant _THIRD_TOKEN_ID_TO_SET = 0xf00df00df00df00df00df00df00df00df00df00df00df00df00df00df00df00d;

function setMultipleDataForSingleTokenId(
    ILSP8 lsp8Contract,
    bytes memory firstTokenIdLsp4MetadataValue,
    bytes memory secondTokenIdLsp4MetadataValue,
    bytes memory thirdTokenIdLsp4MetadataValue
) {
    bytes32[] memory tokenIdsToUpdate = new bytes32[](3);
    bytes32[] memory dataKeysToSet = new bytes32[](3);
    bytes[] memory dataValuesToSet = new bytes[](3);

    tokenIdsToUpdate[0] = _FIRST_TOKEN_ID_TO_SET;
    tokenIdsToUpdate[1] = _SECOND_TOKEN_ID_TO_SET;
    tokenIdsToUpdate[2] = _THIRD_TOKEN_ID_TO_SET;

    // we are setting the metadata for 3 x different tokenIds
    dataKeysToSet[0] = _LSP4_METADATA_KEY;
    dataKeysToSet[1] = _LSP4_METADATA_KEY;
    dataKeysToSet[2] = _LSP4_METADATA_KEY;

    dataValuesToSet[0] = firstTokenIdLsp4MetadataValue;
    dataValuesToSet[1] = secondTokenIdLsp4MetadataValue;
    dataValuesToSet[2] = thirdTokenIdLsp4MetadataValue;

    lsp8Contract.setDataBatchForTokenIds(
        tokenIdsToUpdate,
        dataKeysToSet,
        dataValuesToSet
    );
}
```

> 

Reviewer: ```suggestion
  </TabItem>
</Tabs>
```

Author: There is already a `</TabItem>` tag below, with an other tab opened for ethers v6 after.
